new appWithTranslation's return type not backwards compatible

## 🐛 Bug Report

In my NextJs(pages) app I currently have "i18next": "23.14.0", "next-i18next": "15.3.1", "react-i18next": "15.0.1", and everything works fine.
Doing deps update of i18next to "i18next": "^24.2.2", "next-i18next": "^15.4.1", "react-i18next": "^15.4.0", and getting a type error in my _app.tsx:
<img width="1124" alt="Image" src="https://github.com/user-attachments/assets/e333835c-c717-4175-b50c-4075a278b2d4" />
After spending some time and looking at the code changes in next-i18next I didn't see any obvious changes that would lead to this.
By rolling back the problem went away.

## To Reproduce

A [minimal reproducible example](https://github.com/MadaShindeInai/i18issuerepro).
1. clone repo and write `pnpm i`
2. open pages/_app.tsx and see and error

## Expected behavior

Ensure that the return type of appWithTranslation is backwards compatible or provide instructions on how to migrate correctly to mitigate the TypeScript inconsistency.

## Your Environment

- *runtime version*: node v22.13.0
- *i18next version*: "i18next": "^24.2.2", "next-i18next": "^15.4.1", "react-i18next": "^15.4.0"
- *os*: Mac
- *any other relevant information* "next": "^15.0.1"(same with latest), "react": "^18.3.1" (same with 19+), "@trpc/*": "^11.0.0-rc.730" (same with v10 stable), "@types/eslint": "^8.56.10", (same with 9+), "typescript": "^5.5.3"(same with 5.7.3)

`appWithTranslation` return type is `any`:

<img width="683" alt="Image" src="https://github.com/user-attachments/assets/4f3f9c54-7315-49ba-a849-1b9bc257d90b" />


The `any` type seems caused by missing types for `hoist-non-react-statics` dependency.

```sh
pnpm tsc --noEmit --skipLibCheck false
```

produces an error:

```text
node_modules/.pnpm/next-i18next@15.4.1_i18next@24.2.2_typescript@5.7.3__next@15.1.6_react-dom@18.3.1_react_81ce81a0d8f13b92f2448f2df89683fe/node_modules/next-i18next/dist/types/appWithTranslation.d.ts:2:34 - error TS7016: Could not find a declaration file for module 'hoist-non-react-statics'. '/i18issuerepro-master/node_modules/.pnpm/hoist-non-react-statics@3.3.2/node_modules/hoist-non-react-statics/dist/hoist-non-react-statics.cjs.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/hoist-non-react-statics` if it exists or add a new declaration (.d.ts) file containing `declare module 'hoist-non-react-statics';`

2 import hoistNonReactStatics from 'hoist-non-react-statics';
```

---

Running 

```sh
pnpm add -D @types/hoist-non-react-statics
```

solves the issue in `_app.tsx`:

<img width="606" alt="Image" src="https://github.com/user-attachments/assets/d78d0ddd-21b0-4a3d-9ff4-e4c2d2132bc4" />

--- a/package.json
+++ b/package.json
@@ -92,7 +92,6 @@
     "@size-limit/webpack": "^10.0.2",
     "@size-limit/webpack-why": "^10.0.2",
     "@testing-library/react": "^14.0.0",
-    "@types/hoist-non-react-statics": "^3.3.6",
     "@types/jest": "^29.5.7",
     "@types/node": "^20.8.10",
     "@types/react": "^18.2.33",
@@ -133,6 +132,7 @@
     "webpack": "^5.89.0"
   },
   "dependencies": {
+    "@types/hoist-non-react-statics": "^3.3.6",
     "@babel/runtime": "^7.23.2",
     "core-js": "^3",
     "hoist-non-react-statics": "^3.3.2",

